tuning/autotune.py

```diff
@@ -51,11 +52,58 @@ class CandidateTracker:
     unet_benchmark_time: Optional[float] = None
 
 
-def parse_devices(devices_str: str) -> list[int]:
-    """Parse a comma-separated list of device IDs (e.g., "1,3,5" -> [1, 3, 5])."""
-    devices = []
+def extract_driver_names(user_devices: list[str]) -> set[str]:
+    """Extract driver names from the user devices"""
+    return {device.split("://")[0] for device in user_devices}
+
+
+def fetch_available_devices(drivers: list[str]) -> list[str]:
+    """
+    Extract all available devices on the user's machine for the provided drivers
+    Only the user provided drivers will be queried
+    """
+    all_device_ids = []
+
+    for driver_name in drivers:
+        try:
+            driver = ireert.get_driver(driver_name)
+            devices = driver.query_available_devices()
+            all_device_ids.extend(
+                f"{driver_name}://{device['path']}" for device in devices
+            )
+        except ValueError as e:
+            handle_error(
+                condition=True,
+                msg=print(f"Could not initialize driver {driver_name}: {e}"),
```

Comment:
Why are we printing the message?

Reply:
When I removed the old except statement I forgot to get rid of the print after moving everything to handle_error. I will get it fixed, thanks for pointing it out!